pyfar/plot/_line.py

```diff
@@ -152,19 +152,34 @@ def _time(signal, dB=False, log_prefix=20, log_reference=1,
         ymin = ymax - 90
         ymax = ymax + 10
 
+    # auto detect the time unit
+    if unit is None:
+        unit = _time_auto_unit(signal.times[..., -1])
```

Comment:
Suggestion: I would implement a function which handles the setting of the plot times array, as well as the unit string. The current implementation results in too much redundant code.

Reply:
The reason for this is that `_time_auto_unit()` is also used by the `_group_delay()` plot. If we remove redundancy here we would add it there. Leave it like it is?

Reply:
In this case we would need a second function for handling units in the `_group_delay()` plot, which would add redundancy at a different spot. In the current form, we can separately test the auto uniting, which is reused and have separate code for handling the underlying data, which differs across the plot functions.

Leave it like this, or do I miss what you ment?